spread-tests/regression/lp-1599608/task.yaml

```diff
@@ -0,0 +1,57 @@
+# The setup for this test is unorthodox because by the time the cgroup code
+# is executed, the mounts are in place and /lib/udev/snappy-app-dev from the
+# OS snap is used. Unfortunately, simple bind mounts over
+# /snap/ubuntu-core/current/lib/udev don't work and the OS snap must be
+# unpacked, lib/udev/snappy-app-dev modified to be tested, repacked and
+# mounted. We unmount the OS snap and move it aside to avoid both the original
+# and the updated OS snap from being mounted on the same mount point, which
+# confuses the kernel.
+summary: Check that execle doesn't regress
+# This is blacklisted on debian because we first have to get the dpkg-vendor patches
+systems: [-debian-8]
+restore: |
+    echo Remove hello-world
+    snap remove hello-world
+    echo Unmount the modified OS snap
+    # all of this ls madness can go away when we have remote environment
+    # variables
+    umount $(ls -1d /snap/ubuntu-core/* | grep -v current | tail -1)
+    if [ "x"$(ls -1 /var/lib/snapd/snaps/ubuntu-core_*.snap.orig | tail -1) != "x" ]; then mv -f $(ls -1 /var/lib/snapd/snaps/ubuntu-core_*.snap.orig | tail -1) $(ls -1 /var/lib/snapd/snaps/ubuntu-core_*.snap.orig | tail -1 | sed 's/.orig//') ; fi
+    echo Mount the original OS snap
+    mount $(ls -1 /var/lib/snapd/snaps/ubuntu-core_*.snap | tail -1) $(ls -1d /snap/ubuntu-core/* | grep -v current | tail -1)
+    rm -rf /squashfs-root
+    rm -f /run/spread-test.out
+    rm -f /etc/udev/rules.d/70-spread-test.rules
+    udevadm control --reload-rules
+    udevadm settle
+    udevadm trigger
+    udevadm settle
+execute: |
```

Comment:
Can you please spit this into prepare/execute/restore parts so that it is somewhat more readable